### PR TITLE
[kjobctl] Make kubectl-kjob binaries statically linked.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Make kubectl-kjob binaries statically linked.

#### Which issue(s) this PR fixes:

Part of #2801

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: cmd/experimental/kjobctl/LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (mimowo) [inline cmd/experimental/kjobctl/LICENSE:1]: How is this related to the PR? Can we move to a separete one?

Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/LICENSE:1]: When I running `make artifacts` I have an error that `LICENSE` file not found on `multiplatform-build.sh`.

File: cmd/experimental/kjobctl/LICENSE
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/LICENSE:1]: As far as I know we will move `kjobctl` to another repository. So we can't use `kueue` `LICENSE`.

File: cmd/experimental/kjobctl/LICENSE (hunk 1 above)
Comment (mimowo) [reply cmd/experimental/kjobctl/LICENSE:1]: Can we move it to a preparatory PR?

Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/LICENSE]: #2819

**Review by trasc — COMMENTED** [review]
/lgtm
/approve
